pkg/lib/aws/ecr.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"encoding/base64"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/cortexlabs/cortex/pkg/lib/errors"
+)
+
+type ECRAuthConfig struct {
+	Username      string
+	AccessToken   string
+	ProxyEndpoint string
+}
+
+func GetECRAuthToken() (*ecr.GetAuthorizationTokenOutput, error) {
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	svc := ecr.New(sess)
+	input := &ecr.GetAuthorizationTokenInput{}
+	result, err := svc.GetAuthorizationTokenWithContext(context.Background(), input)
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ecr.ErrCodeServerException:
+				return result, errors.Wrap(aerr, ecr.ErrCodeServerException, "failed to retrieve ECR auth token")
+			case ecr.ErrCodeInvalidParameterException:
+				return result, errors.Wrap(
+					aerr,
+					ecr.ErrCodeInvalidParameterException,
+					"failed to retrieve ECR auth token",
+				)
+			default:
```

Comment:
Good thoughts. Since you only care whether the get authorization function has failed or not, it's just better to return the result and the wrapped error.